pkg/collector/cluster.go

```diff
@@ -127,3 +132,14 @@ func (c *ClusterCollector) Get() ([]map[string]interface{}, error) {
 
 	return results, nil
 }
+
+func (c *ClusterCollector) getLastAppliedConfig(r unstructured.Unstructured) (string, bool) {
```

Comment:
Can we add tests for this new function? I guess it will require a similar approach with the fake client as [^1]. I would suggest testing that we get:
- `kubectl.kubernetes.io/last-applied-configuration` if present
- false if none present
- custom additional annotation if requested and present
- and that kubectl one takes precedence (this would cover the issue discussed below)

[^1]: https://github.com/doitintl/kube-no-trouble/blob/a339b08f2e91c142e2dd80026e1039d343033b87/pkg/collector/cluster_test.go#L104



[1]: https://github.com/doitintl/kube-no-trouble/blob/a339b08f2e91c142e2dd80026e1039d343033b87/pkg/collector/cluster_test.go#L104

Reply:
Thinking about this a bit more, resp. about the difficult testing. One though - what if we rewrite the `getLastAppliedConfig` as:
```go
func (c *ClusterCollector) getLastAppliedConfig(annotations []string) (string, bool) {
```

and then call it as:
```go
...
if jsonManifest, ok := c.getLastAppliedConfig(r.GetAnnotations()); ok {
```

I think it would make testing much easier.

Reply:
_It might still be easier for testing to refactor to:_
```go
func (c *ClusterCollector) getLastAppliedConfig(resourceAnnotations []string) (string, bool) {
```
_but either is fine._